### PR TITLE
Make transpose, trace, kron, reshape, sum op support complex type

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

New features
### PR changes

OPs
### Describe

1. make transpose, trace, kron, reshape, sum op support complex type
2. fix the bug of imag part of complex not initialized

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_complex_sum_layer.py [hunk 1]
```diff
@@ -37,6 +38,21 @@ def test_complex_x(self):
                 target = np.sum(input, axis=(1, 2))
                 self.assertTrue(np.allclose(result, target))
 
+    def test_complex_basic_api(self):
+        input = rand([2, 10, 10]).astype(self._dtype) + 1j * rand(
```
Comment (chenwhql) [inline python/paddle/fluid/tests/unittests/test_complex_sum_layer.py]: make sure our test cases include `[float32, float64] * [CPU, GPU]`, `real (op) complex` and `complex (op) real`, the original unittests for complex are incomplete

Comment (MingMingShangTian) [reply python/paddle/fluid/tests/unittests/test_complex_sum_layer.py]: done, thanks.

File: python/paddle/fluid/tests/unittests/test_complex_reshape.py [hunk 2]
```diff
@@ -13,38 +13,80 @@
...
+                                                              4).astype(dtype)
+            shape = (2, -1)
+            for place in self._places:
+                with dg.guard(place):
+                    x_var = dg.to_variable(x_np)
+                    y_var = cpx.reshape(x_var, shape)
+                    y_np = y_var.numpy()
+                    np.testing.assert_allclose(np.reshape(x_np, shape), y_np)
+
+    def test_case2(self):
```
Comment (chenwhql) [inline python/paddle/fluid/tests/unittests/test_complex_reshape.py:43]: we don't recommend naming by `1,2,3,4...`, the original code style is not good, maybe you can polish test_case's name in next pr

**Review by chenwhql — APPROVED** [review]
LGTM, comments can be polished in next PR

File: python/paddle/fluid/tests/unittests/test_complex_reshape.py (hunk 2 above)
Comment (MingMingShangTian) [reply python/paddle/fluid/tests/unittests/test_complex_reshape.py:43]: ok, I will change the test_case's name in next pr.

**Pull request merged**